Packs/Slack/Integrations/Slack/Slack.py

```diff
@@ -857,39 +719,13 @@ def extract_entitlement(entitlement: str, text: str) -> Tuple[str, str, str, str
 
 
 async def slack_loop():
-    """
-    Starts a Slack RTM client while checking the connection.
-    """
-    while True:
-        loop = asyncio.get_running_loop()
-        rtm_client = None
-        try:
-            rtm_client = slack.RTMClient(
-                token=BOT_TOKEN,
-                run_async=True,
-                loop=loop,
-                auto_reconnect=False,
-                proxy=PROXY_URL,
-                ssl=SSL_CONTEXT
-            )
-
-            client_future = rtm_client.start()
-            while True:
-                await asyncio.sleep(10)
-                if rtm_client._websocket is None or rtm_client._websocket.closed or client_future.done():
-                    ex = client_future.exception()
-                    if ex:
-                        demisto.error(f'Slack client raised an exception: {ex}')
-                    demisto.info('Slack - websocket is closed or done')
-                    break
-        except Exception as e:
-            error = f'Slack client raised an exception: {e}'
-            await handle_listen_error(error)
-        finally:
-            # If we got here, the websocket is closed or the client can't connect. Will try to connect every 5 seconds.
-            if rtm_client and not rtm_client._stopped:
-                rtm_client.stop()
-            await asyncio.sleep(5)
+    client = SocketModeClient(
+        app_token=APP_TOKEN,
+        web_client=ASYNC_CLIENT
+    )
+    client.socket_mode_request_listeners.append(listen)
+    await client.connect()
+    await asyncio.sleep(float("inf"))
```

Comment:
Do we need asyncio here?

Reply:
Asyncio here is just being used to keep the process alive indefinitely. We could use a `while true` loop or something similar.